src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -437,7 +444,14 @@ private bool HasHome(BoundFieldAccess fieldAccess, bool needWriteable)
                 return false;
             }
 
-            if (!needWriteable && !EnablePEVerifyCompat())
+            // when reference is strictly required, consider fields as addressable
```

Comment:
I don't think this comment says what you mean. Even if we copy into a temp, you're going to get an address back. Really, this is a statement about not being allowed to copy.